docs/proposals/scheduling/20220701-schedule-gang.md

```diff
@@ -0,0 +1,471 @@
+will increase the risk of resource deadlock between Gangs. For example, the quota of the quota group is 10(quota will be proposed later), 
+and the user submits three Gangs with 5 pods. Due to various plugin constraints, Gang1\2\3 may allocate resources of 3\3\4 respectively. 
+Since the quota group's quota is full, there will be no new resource scheduling. We call this is resource deadlock of resource Gang.
+In future proposal, we will try to fix this problem.
+
+#### GangGroup
+As mentioned above, Some jobs have complex Gang requirements. For example, a job has several roles. Each role will have several pods 
+and its own Gang conditions. Jobs also need different roles to form different GangGroups. All pods in a GangGroup can 
+trigger the bind process only after all roles in a GangGroup meet their Gang conditions. So we introduce `GangGroup` concept,
+which allow user to bundle different Gangs together.
+
+#### After Gang
+It should be noted that, if the resource accumulation conditions of Gang are met, then some pods failed in the process of binding,
+or some bound pods are preempted\rescheduled, should the constraints of Gang still be effective in the process of resource reallocation? 
+Because the initial purpose of Gang is to require pods to be pulled up at the same time, if some pods have been pulled up, 
+then the subsequent Gang behavior is meaningless. Therefore, when once Gang has been satisfied, all subsequent resource allocations 
+are no longer constrained by Gang rules, and their performance is similar to ordinary pod.
+
+As mentioned above, `WaitTime` is the max wait time since first pod comes to permit stage. if `WaitTime` is timeout, 
+scheduler will roll back all assumed pods, update each pod's annotation with `gang.scheduling.koordinator.sh/timeout=true`, and
+won't schedule these pods anymore. User should pay attention to this status and delete pods timely.
+
+### API
+#### Definition
+
+Our original intention is to improve and enhance the ability of the community's original `PodGroup`, so we will be 
+compatible with the way the community declares the `PodGroup`.
+
+We recommend users declaring Gang scheduling configuration by pod's annotation. First Reason, high level operator has no need 
+to maintain Gang crd life circle like handling `update/create/delete` events. Second Reason, from a Scheduler perspective, 
+it's inconvenient to maintain receive-order-issue's between Gang crd and pod. For example, if pods arrive to scheduler before Gang crd,
+we have to build a fake Gang data structure temporarily to collect all related pods, and need to suspend the scheduling of pods 
+until parse the configuration from real Gang crd. According to practical experience, pod's annotation is enough for declaring Gang 
+scheduling configuration, so we build the Gang's configuration according to the first pod which declared the Gang.
```

Comment:
It is recommended to adjust the order here:
1、first describe the user story
2、describe the extension of annotation on the PodGroup to meet the needs of these scenarios
3、finally describe the way of annotation and what its benefits are. But don't make recommendations in proposals, this tendency is too obvious.

Reply:
fixed